server/src/handlers/bookmarks.rs

```diff
@@ -28,19 +28,32 @@ async fn create_bookmark(
         .map_err(|_| StatusCode::INTERNAL_SERVER_ERROR)
 }
 
+async fn delete_bookmark(
+    user: User,
+    State(db): State<DB>,
+    Json(bookmark): Json<Bookmark>,
```

Comment:
Once you've read the bookmark ID from the path of the request, there is actually no need to have the rest of the bookmark in the body as `json`. Indeed, the GUI in its current state does not send the entire bookmark for deletion, it only includes its ID in the path of the request.